src/benchmark_register.cc

```diff
@@ -202,7 +202,8 @@ bool FindBenchmarksInternal(const std::string& re,
 Benchmark::Benchmark(const char* name)
     : name_(name),
       aggregation_report_mode_(ARM_Unspecified),
-      time_unit_(kNanosecond),
+      time_unit_(GetDefaultTimeUnit()),
+      use_default_time_unit_(true),
```

Comment:
i don't understand why you need this boolean.

if you initialise to the default, and then the user overrides it by calling `Unit`, then you can just return `time_unit_` knowing it's either the default or the override.  what is this boolean protecting against?

Reply:
If you use the predefined macros `BENCHMARK`, `BENCHMARK_F` then the `Benchmark` objects are created by global initialization so argument parsing and calling `SetDefaultTimeUnit` (or even `main`) will be done later.

Reply:
ah very well spotted, thanks :)